File: csharp/src/Behavioral/Strategy/DvdNameTheAtEndStrategy.cs
//DvdNameTheAtEndStrategy.java - two of three concrete strategies


namespace Behavioral.Strategy {

public class DvdNameTheAtEndStrategy : DvdNameStrategy {
   override public string formatDvdName(string dvdName, char charIn) {
       if (dvdName.StartsWith("The ")) {
           return dvdName.Substring(4) + ", The";
       }
       else if (dvdName.StartsWith("THE ")) {
           return dvdName.Substring(4) + ", THE";
       }       
       else if (dvdName.StartsWith("the ")) {
           return dvdName.Substring(4) + ", the";
       }         
       return dvdName;
   }
}

}

File: scala/src/Behavioral/Strategy/DvdNameTheAtEndStrategy.scala
package Behavioral.Strategy;


//DvdNameTheAtEndStrategy - two of three concrete strategies

public class DvdNameTheAtEndStrategy extends DvdNameStrategy {
   public String formatDvdName(String dvdName, char charIn) {
       if (dvdName.startsWith("The ")) {
           return new String(dvdName.substring(4, 
             (dvdName.length())) + ", The");
       }
       if (dvdName.startsWith("THE ")) {
           return new String(dvdName.substring(4, 
             (dvdName.length())) + ", THE");
       }       
       if (dvdName.startsWith("the ")) {
           return new String(dvdName.substring(4, 
             (dvdName.length())) + ", the");
       }         
       return dvdName;
   }
}